modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -1214,6 +1231,46 @@ else if (spiState == LOOPBACK_PROBLEM) {
             .end();
     }
 
+    /**
+     * Schedules a maintenance task to clear cache folders based on the exception message.
+     *
+     * @param chackFailedMsg Exception message.
+     */
+    private void scheduleMaintenanceTaskToClearCacheFolders(TcpDiscoveryCheckFailedMessage chackFailedMsg) {
+        ArrayList<CacheConfiguration> cacheCfgs = new ArrayList<>();
+
+        for (String cacheName : chackFailedMsg.error()
+            .replaceFirst("^.*\\[", "")
+            .replaceFirst("\\].*", "")
+            .split(", ")) {
+            CacheConfiguration cc = gridKernalContext().cache().context().cacheContext(CU.cacheId(cacheName)).config();
+
+            cacheCfgs.add(cc);
+        }
+
+        FilePageStoreManager pageStore = (FilePageStoreManager) gridKernalContext().cache().context().pageStore();
+
+        try {
+            String params = cacheCfgs.stream()
+                .map(ccfg -> pageStore.cacheWorkDir(ccfg).getName())
+                .collect(Collectors.joining(File.separator));
+
+            log.info("Task params: " + params + " sep: " + File.separator);
+
+            gridKernalContext().maintenanceRegistry()
+                .registerMaintenanceTask(
+                    new MaintenanceTask(
+                        ClearFolderWorkflow.CLEAR_FOLDER_TASK,
+                        "Corrupted cache groups are already removed from other nodes",
```

Comment:
Misleading description of Maintenance Task - this situation isn't necessarily related to cache corruption. Please rewrite it in more general form.